### DISCUSSION TITLE
Liquidsoap Integration?

### DISCUSSION THREAD

**Discussion by Aragur** [snapcast/snapcast#1034] [General]
Hey I'm pretty stuck right now.
I have a live stream via icecast, now I realised that multiroom setups aren't possible using this because each node has its own delay.
Currently I'm using [Liquidsoap](https://www.liquidsoap.info/) to switch my sources with a nice transition.  
Liqudsoap supports direct export to Icecast and Shoutcast. Is it also possible to use Liquidsoap as a input source for snapcast?

**Comment by artmg** [comment]
https://www.liquidsoap.info/doc-2.1.0/ mentions that ALSA is an output option, and you can use an ALSA loopback to get that sound into snapserver: https://github.com/badaix/snapcast/blob/master/doc/configuration.md#alsa

You should configure output from Liquidsoap to go into one ALSA device on the Loopback Card (e.g. plughw:0,1,0) and tell Snapserver to look on the next device (e.g. `source = alsa://?name=SomeName&device=hw:0,1,0`) as that's how ALSA loopbacks work.
